# Endophytes_wildgrassplants
# Optimised Workflow for Detecting Root Endophytic Bacterial Communities
This repository contains scripts used in the study:
# “Unveiling Hidden Endophytes by Optimising Identification of Endophytic Bacterial Communities from Wild Grassland Plant Roots.”
The project provides a reproducible computational workflow for processing, analysing, and visualising 16S rRNA sequencing data generated to evaluate protocol optimisation for detecting bacterial endophytes in plant roots.

\n This project uses a reproducible bioinformatics pipeline to process 16S rRNA amplicon sequencing data for identifying bacterial endophytic communities from plant roots. The workflow integrates QIIME2 processing, downstream analysis in R using phyloseq, and comparative analyses with and without host-derived sequences.



The scripts cover:
Raw sequence quality control
Amplicon processing and ASV inference
Removal of host-derived sequences
Taxonomic classification
Diversity analysis
Statistical comparisons
Visualisation of microbial communities
